mercury/tests/test_configure_sensors.py

```diff
@@ -100,3 +128,207 @@ def test_ConfigureSensorView_POST_success_correct_object_created(self):
             sensor.sensor_format[self.field_name_2]["unit"],
             self.test_sensor[self.field_name_2]["field_unit"],
         )
+
+    # Invalid POST tests
+
+    # Duplicate field names still returns status ok
+    def test_ConfigureSensorView_invalid_POST_duplicate_fields_returns_status_ok(self):
+        # Login
+        self._get_with_event_code(self.sensor_url, self.TESTCODE)
+
+        # POST sensor data
+        response = self.client.post(
+            reverse(self.sensor_url),
+            data={
+                "sensor-name": self.test_sensor["name"],
+                "field-name": [self.field_name_1, self.field_name_1],
+                "field-type": [
+                    self.test_sensor[self.field_name_1]["field_type"],
+                    self.test_sensor[self.field_name_2]["field_type"],
+                ],
+                "field-unit": [
+                    self.test_sensor[self.field_name_1]["field_unit"],
+                    self.test_sensor[self.field_name_2]["field_unit"],
+                ],
+            },
+        )
+
+        # Check that POST redirects to sensor (same page reloads)
+        self.assertEqual(200, response.status_code)
+
+    # Duplicate field names - no AGSensor object created
+    def test_ConfigureSensorView_invalid_POST_field_duplicates_no_object_created(self):
+        # Login
+        self._get_with_event_code(self.sensor_url, self.TESTCODE)
+
+        # POST sensor data with duplicate field names
+        self.client.post(
+            reverse(self.sensor_url),
+            data={
+                "sensor-name": self.test_sensor["name"],
+                "field-name": [self.field_name_1, self.field_name_1],
+                "field-type": [
+                    self.test_sensor[self.field_name_1]["field_type"],
+                    self.test_sensor[self.field_name_2]["field_type"],
+                ],
+                "field-unit": [
+                    self.test_sensor[self.field_name_1]["field_unit"],
+                    self.test_sensor[self.field_name_2]["field_unit"],
+                ],
+            },
+        )
+
+        # Check that AGSensor object is created in db with expected params
+        sensors = AGSensor.objects.all()
+        self.assertEqual(sensors.count(), 0)
+
+    # Sensor name missing still returns status OK
+    def test_ConfigureSensorView_invalid_POST_sensor_name_missing_status_ok(self):
+        # Login
+        self._get_with_event_code(self.sensor_url, self.TESTCODE)
+
+        # POST sensor data
+        response = self.client.post(
+            reverse(self.sensor_url),
+            data={
+                "sensor-name": "",
+                "field-name": [self.field_name_1, self.field_name_1],
```

Comment:
Not sure if it matters, but should you still be having the same field names when testing for missing name?

Reply:
no this was a mistake